src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -102,11 +106,44 @@ public override Conversion GetMethodGroupDelegateConversion(BoundMethodGroup sou
             }
 
             var resolution = ResolveDelegateOrFunctionPointerMethodGroup(_binder, source, methodSymbol, isFunctionPointer, callingConventionInfo, ref useSiteInfo);
+            if (resolution.IsExtensionMember(out var extensionMember))
+            {
+                var nestedConversion = ClassifyConversionFromExpressionType(extensionMember.GetTypeOrReturnType().Type, destination, isChecked: false, ref useSiteInfo);
+                if (nestedConversion.Kind != ConversionKind.NoConversion)
+                {
+                    return new Conversion(extensionMember, nestedConversion);
+                }
+            }
+
             var conversion = (resolution.IsEmpty || resolution.HasAnyErrors) ?
                 Conversion.NoConversion :
                 ToConversion(resolution.OverloadResolutionResult, resolution.MethodGroup, methodSymbol.ParameterCount);
             resolution.Free();
             return conversion;
+
+            bool tryGetExtensionMemberConversion(BoundMethodGroup source, TypeSymbol destination,
+                ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo, out Conversion extensionMemberConversion)
+            {
+                if (source is not { Methods: [], SearchExtensionMethods: true })
+                {
+                    extensionMemberConversion = default;
+                    return false;
+                }
+
+                MethodGroupResolution resolution = _binder.ResolveMethodGroup(source, analyzedArguments: null, isMethodGroupConversion: false, ref useSiteInfo);
```

Comment:
>_binder [](http://example.com/codeflow?start=51&length=7)

I think should use binder that we store in ```BoundMethodGroup```. I do not think there is a guarantee that the binder stored in the Conversions is the same that was used to bind the group.

Reply:
On the other hand we use it for ```ResolveDelegateOrFunctionPointerMethodGroup```, it is quite possible there is no difference.

Reply:
The binder that is passed into the `BoundMethodGroup` constructor doesn't get stored. It just gets used to compute `GetFunctionType`.

Reply:
> The binder that is passed into the BoundMethodGroup constructor doesn't get stored. 

But we could store it if that is actually necessary. The question is: "Is that necessary?"

Reply:
For example, could accessibility check be affected somehow, etc.?

Reply:
Are we using the binder to figure out extension scopes? Is this binder always gives us the right scopes? I would prefer us not making too many assumptions about the binder stored in Conversions

Reply:
For example, we store context binder in UnboundLambda, and probably use it when we classy conversions from the lambda